iele-assemble/src/IeleDesugar.hs

```diff
@@ -178,3 +183,11 @@ compileContracts (c:cs) = go Map.empty c cs
                          (assemble (compileContract childContracts c))
                          childContracts)
              c' cs
+
+namedSourceMaps :: [ContractP] -> String
+namedSourceMaps cs =
+  let funcInstructions f = (functionDefinitionEntry f)++(concatMap labeledBlockInstructions $ functionDefinitionBlocks f) in
+  let allContractInstructions c = concatMap funcInstructions $ functionDefinitions c in
+  let contractSourceMap c = intercalate ";" $ map (show . info) (allContractInstructions c) in
+  let namedSourceMaps (IeleNameText name,c) = concat [name,",",contractSourceMap c] in
+  intercalate "\n" $ map (namedSourceMaps . processContract) cs
```

Comment:
The ordinary way to write this would be as a single let-block:

```suggestion
  let funcInstructions f = (functionDefinitionEntry f)++(concatMap labeledBlockInstructions $ functionDefinitionBlocks f)
      allContractInstructions c = concatMap funcInstructions $ functionDefinitions c
      contractSourceMap c = intercalate ";" $ map (show . info) (allContractInstructions c)
      namedSourceMaps (IeleNameText name,c) = concat [name,",",contractSourceMap c]
   in intercalate "\n" $ map (namedSourceMaps . processContract) cs
```

Is there a reason not to do that here?

Reply:
Nope, I just haven't used Haskell enough.